src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/AppealServiceTest.kt

```diff
@@ -0,0 +1,194 @@
+        "Some information about the decision made",
+        application,
+        createdByUser,
+      )
+
+      assertThat(result).isInstanceOf(AuthorisableActionResult.Success::class.java)
+      result as AuthorisableActionResult.Success
+      assertThat(result.entity).isInstanceOf(ValidatableActionResult.FieldValidationError::class.java)
+      val resultEntity = result.entity as ValidatableActionResult.FieldValidationError
+      assertThat(resultEntity.validationMessages).containsEntry("$.reviewer", "empty")
+    }
+
+    @ParameterizedTest
+    @EmptySource
+    @ValueSource(strings = ["  ", "\t", "\n"])
+    fun `Returns FieldValidationError if the decision detail is blank`(decisionDetail: String) {
+      createdByUser.addRoleForUnitTest(UserRole.CAS1_APPEALS_MANAGER)
+
+      val result = appealService.createAppeal(
+        LocalDate.now(),
+        "Some information about why the appeal is being made",
+        "ReviewBot 9000",
+        AppealDecision.accepted,
+        decisionDetail,
+        application,
+        createdByUser,
+      )
+
+      assertThat(result).isInstanceOf(AuthorisableActionResult.Success::class.java)
+      result as AuthorisableActionResult.Success
+      assertThat(result.entity).isInstanceOf(ValidatableActionResult.FieldValidationError::class.java)
+      val resultEntity = result.entity as ValidatableActionResult.FieldValidationError
+      assertThat(resultEntity.validationMessages).containsEntry("$.decisionDetail", "empty")
+    }
+
+    @Test
+    fun `Stores appeal in repository and returns the stored appeal`() {
+      createdByUser.addRoleForUnitTest(UserRole.CAS1_APPEALS_MANAGER)
+
+      val now = LocalDate.now()
+
+      every { appealRepository.save(any()) } returnsArgument 0
+
+      val result = appealService.createAppeal(
+        now,
+        "Some information about why the appeal is being made",
+        "ReviewBot 9000",
+        AppealDecision.accepted,
+        "Some information about the decision made",
+        application,
+        createdByUser,
+      )
+
+      assertThat(result).isInstanceOf(AuthorisableActionResult.Success::class.java)
+      result as AuthorisableActionResult.Success
+      assertThat(result.entity).isInstanceOf(ValidatableActionResult.Success::class.java)
+      val resultEntity = result.entity as ValidatableActionResult.Success
+      assertThat(resultEntity.entity).matches {
+        it.appealDate == now &&
+          it.appealDetail == "Some information about why the appeal is being made" &&
+          it.reviewer == "ReviewBot 9000" &&
+          it.application == application &&
+          it.createdBy == createdByUser
+      }
+      verify(exactly = 1) {
+        appealRepository.save(
+          match {
+            it.appealDate == now &&
+              it.appealDetail == "Some information about why the appeal is being made" &&
+              it.reviewer == "ReviewBot 9000" &&
+              it.application == application &&
```

Comment:
do we also want to verify that the expected decision is being recorded by the repository? e.g. something like:

```
it.decision == AppealDecision.accepted
```

Reply:
Good catch, I've updated the test to include the `decision` and the `decisionDetail` fields.